src/integration/setup.ts

```diff
@@ -0,0 +1,105 @@
+import { IndexList, Pinecone } from '../index';
+import {
+  diffPrefix,
+  generateRecords,
+  globalNamespaceOne,
+  prefix,
+  sleep,
+} from './test-helpers';
+
+// todo: refactor to make conditions & loops more efficient
+
+const setup = async () => {
```

Comment:
nit: I'd maybe add comments to the setup and teardown functions that call out the need for passing `SERVERLESS_INDEX_NAME` through environment variables, just so it's clear.